wasm/src/language_service.rs

```diff
@@ -178,6 +208,47 @@ serializable_type! {
     IDefinition
 }
 
+serializable_type! {
+    SignatureHelp,
+    {
+        signatures: Vec<SignatureInformation>,
+        active_signature: u32,
+        active_parameter: u32,
+    },
+    r#"export interface ISignatureHelp {
+        signatures: ISignatureInformation[];
+        active_signature: number;
+        active_parameter: number;
+    }"#,
+    ISignatureHelp
+}
+
+serializable_type! {
+    SignatureInformation,
+    {
+        label: String,
+        documentation: Option<String>,
+        parameters: Vec<ParameterInformation>,
+    },
+    r#"export interface ISignatureInformation {
+        label: string;
+        documentation: string | undefined;
+        parameters: IParameterInformation[];
+    }"#
+}
+
+serializable_type! {
+    ParameterInformation,
+    {
+        label: Span,
+        documentation: Option<String>,
+    },
+    r#"export interface IParameterInformation {
+        label: { start: number; end: number };
+        documentation: string | undefined;
```

Comment:
You can use `ISpan` here which is now defined in this file. Also `?` for an optional parameter looks more consistent in style (though it's basically the same as `string | undefined`)

```suggestion
        label: ISpan;
        documentation?: string;
```

Reply:
done